test/honest_test.go

```diff
@@ -71,14 +71,14 @@ func TestHonest_Agreement(t *testing.T) {
 			name := fmt.Sprintf("%s %d", test.name, participantCount)
 			t.Run(name, func(t *testing.T) {
 				t.Parallel()
-				test.options = append(test.options,
+				opts := append(test.options,
```

Comment:
This append will is not guaranteed to make a copy. It could append onto the existing capacity in `test.options`.

Probably do:
```
opts := append([]sim.Option{}, test.options...)
opts = append(opts, ...)
```